packages/enclave-contracts/contracts/Enclave.sol

```diff
@@ -324,9 +350,39 @@ contract Enclave is IEnclave, OwnableUpgradeable {
         );
         require(success, InvalidOutput(plaintextOutput));
 
+        _distributeRewards(e3Id);
+
         emit PlaintextOutputPublished(e3Id, plaintextOutput);
     }
 
+    ////////////////////////////////////////////////////////////
+    //                                                        //
+    //                   Internal Functions                   //
+    //                                                        //
+    ////////////////////////////////////////////////////////////
+
+    function _distributeRewards(uint256 e3Id) internal {
+        IRegistryFilter.Committee memory committee = ciphernodeRegistry
+            .getCommittee(e3Id);
+        uint256[] memory amounts = new uint256[](committee.nodes.length);
+
+        // We might need to pay different amounts to different nodes.
```

Comment:
Make sure to leave "TODO" in the comment so others can easily find things like this that we know will need to be changed in future.